### PR TITLE
reducing serialized space

### PR DESCRIPTION
**Pull request by sudiptoguha**
*Description of changes:* avoiding storing information which is redundant in the forest

### CODE REVIEW

File: Java/core/src/main/java/com/amazon/randomcutforest/state/RandomCutForestState.java [hunk 1]
```diff
@@ -48,8 +50,18 @@
...
+
     private boolean centerOfMassEnabled;
 
+    private boolean saveTreeState;
+
+    private boolean saveSamplerState;
+
+    private boolean saveCoordinatorState;
+
+    private boolean singlePrecisionSet;
```
Comment (jotok) [inline Java/core/src/main/java/com/amazon/randomcutforest/state/RandomCutForestState.java:63]: I would either use the Precision enum or a String value here.

File: Java/core/src/main/java/com/amazon/randomcutforest/store/SmallLeafStore.java [hunk 2]
```diff
@@ -37,20 +37,17 @@
  * The reference to the nodes will be offset by an amount that this ihe
  * capacity.
  */
-public class SmallLeafStore extends SmallIndexManager implements ILeafStore {
+public class SmallLeafStore extends IndexManager implements ILeafStore {
```
Comment (jotok) [inline Java/core/src/main/java/com/amazon/randomcutforest/store/SmallLeafStore.java:40]: Did we want to update these classes as well to has-an IndexManager from is-an Index manager?

Comment (sudiptoguha) [reply Java/core/src/main/java/com/amazon/randomcutforest/store/SmallLeafStore.java:40]: Great question -- we probably should. Given that we had to do that for PointStore and is-an became incredibly cumbersome from a constructor point of view, it is likely that down the road we will have to do the same for NodeStore.
LeafStores may not be needed at all even.

**Comment by sudiptoguha** [comment]
Refactoring and removing leaf store as well as multiple files managing the leafstore.

**Pull request closed without merging**